addons/blocklydevtools/userscript.js

```diff
@@ -0,0 +1,488 @@
+            if (window.getSelection) {
+                const sel = window.getSelection();
+
+                sel.modify("extend", "backward", "paragraphboundary"); //Alter bounds of selection
+
+                const pos = sel.anchorOffset; //Offset of caret in anchorNode (in the editor's case, this is the current Text node)
+
+                if (sel.anchorNode) {
+                    sel.collapseToEnd();
+
+                    const node = sel.anchorNode;
+
+                    if (node instanceof HTMLElement) {
+                        //Get text before and after caret
+                        const preText = node.innerHTML.substring(0, pos);
+                        const postText = node.innerHTML.substring(pos, node.innerHTML.length);
+
+                        node.innerHTML = preText + "&emsp;" + postText; //Insert tab character
+                    } else if (node instanceof Node) {
+                        //Get text before and after caret
+                        const preText = node.nodeValue.substring(0, pos);
+                        const postText = node.nodeValue.substring(pos, node.nodeValue.length);
+
+                        node.nodeValue = preText + "\u2003" + postText; //Insert tab character
+                    }
+
+                    sel.setPosition(node, pos + 1); //Move text caret forward
+                }
+            }
+        },
+        blockKeyEvent: function (keyEvent) {
+            if (keyEvent.preventDefault) {
+                keyEvent.preventDefault();
+            } else {
+                keyEvent.returnValue = false;
+            }
+        }
+    };
+
+    function firefoxSpacingFix(elem) { //Fix for firefox adding a \n at the end of every text node, which makes the editor feel janky.
+        elem.childNodes.forEach(element => {
+            if (element instanceof Text) {
+                element.nodeValue = element.nodeValue.replace(/\n/g, "");
+            }
+        })
+    }
+
+    function makeEditorContainer() {
+        var editor = document.createElement("div");
+
+        editor.setAttribute("data-isblocklydeveditor", "true"); //Attribute to identify element as editor
+        editor.contentEditable = true; //Allow editing of content
+
+        //Styling
+        editor.style.padding = "10px";
+        editor.style.userSelect = "text";
+        editor.style.cursor = "auto";
+        editor.style.font = "12pt monospace";
+        editor.style.boxShadow = "0px 0px 10px rgba(0,0,0,0.5)";
+        editor.style.color = "yellowgreen";
+        editor.style.border = "2px solid white";
+        editor.style.zIndex = "998";
+        editor.style.borderRadius = "0.8rem";
+        editor.style.width = "max-content";
+        editor.style.backgroundColor = "rgb(0,0,30)";
+        editor.style.overflowX = "hidden";
+        editor.style.maxHeight = "100vh";
+        editor.style.overflowY = "scroll";
+        editor.style.caretColor = "white";
```

Comment:
Could these and styles for other elements the addon adds also be moved to the CSS file?

Reply:
Yep, probably.

Reply:
Done.